### DISCUSSION TITLE
Hardware suggestions for a new new build.

### DISCUSSION THREAD

**Discussion by 0x9c7d0d94** [artisan-roaster-scope/artisan#814] [Q&A] [answered]
Hi there, I have recently purchased a second hand STA - 5kg roaster. It needs a bit of a refurb/clean and I plan on stripping it down to do so.
The roaster is a hot air roaster, with the gas burner in the chamber on the side and then a fan to move the hot air into the roasting chamber. The beans sit in a perforated drum for agitation.

Picture below for reference.
![Screenshot_20220128-103959_eBay](https://user-images.githubusercontent.com/96889412/152682248-9a5e2eb1-e9d6-47c3-956a-6d7ab8ffb993.jpg)

So I was wondering if anyone had any advice on the hardware set up required for this roaster.
Would having a 3rd probe, located in the side chamber be a good idea to track that in addition to the BT and ET? 

As for the hardware; what is currently recommended, should I go for phidget, yoctopuce or something else?

In the future I would like to add fan and burner control within artisan, but for now i am happy to just get the temp read outs working, unless that would be recommended to do so now (while it is out of commission/being cleaned up). Not sure if this changes the options i have in regard to control boards etc.

For jow I think I'd just like a shopping list of items I need to buy so I can get everything on the bench and figure the next steps out, this will be my first time converting a roaster for use it artisan but I am semi familiar with some of the components from robotics.

Any input is appreciated.

**Comment by CarefreeBuzzBuzz** [comment, marked as answer]
Hard to know until you start roasting if that third probe would be of value at that location.  
You could start with this Artisan recommended RTD set:
https://shop.artisan.plus/en/hardware/phidgets-2xrtd-set <https://shop.artisan.plus/en/hardware/phidgets-2xrtd-set>

Then after some roasts decide where to go next.   Guessing if its a hot air roaster you might want temps at inlet and outlet.    The VintHub would allow you to add 4 more devices such as another RTD, and phidgets for environmental temp. 

 
----------------------------------------
Michael Herbert, Head Roaster
Order at www.carefreebuzzbuzz.com <http://www.carefreebuzzbuzz.com/>
Artisan RoasterScope  <https://artisan-scope.org/>Team

**Reply by 0x9c7d0d94** [reply]
Great, thank you for the shop link - I had all the components in a shopping basket on another site as I was unaware of the shop :)

I have placed an order for the RTD kit and the ambient set.
Which would leave me 2 more inputs on the hub - I shall try and reach out to other owners of the roaster in regards to the placement as they will probably have a better idea than myself on that.

I have recently been advised to upgrade the gas controls from the standard one as it doesnt offer much control - I am now thinking I should maybe use artisan for this from the get go.
In regards to automating/controlling the gas (LPG) and maybe the fans from artisan; what components would be needed? With only 2 remaining ports on the VINT hub, I'd like to leave those for additional RTD probes if/when needed. Would i need an expansion board, or simply add another hub etc? What would be the preferred gas valve to use?

**Comment by 0x9c7d0d94** [comment]
Ok, so everything turned up from Germany VERY quickly (to the UK). For anyone reading this and concerned about the time frame of the kit getting to you then I wouldnt be concerned at all :)

I was wondering what "best practice" is regarding the environmental phidgets (humidity etc).
Where would you physically place these? The vint cables are not very long so in it's current form I'd need to place them in the control box with the vint hub... I assume this isnt correct?

Would getting long cables and mounting them else where (but still near the roaster) be better, or would using an additional vint hub and mount them further away from the roaster? The last option would give me more ports on the vint hub to maybe control the roaster with in the future?
